roles/postgres_hardening/tasks/hardening.yml

```diff
@@ -0,0 +1,151 @@
+---
+- name: Fetch OS dependent variables
+  ansible.builtin.include_vars:
+    file: "{{ item }}"
+    name: os_vars
+  with_first_found:
+    - files:
+        - "{{ ansible_facts.distribution }}_{{ ansible_facts.distribution_major_version }}.yml"
+        - "{{ ansible_facts.distribution }}.yml"
+        - "{{ ansible_facts.os_family }}_{{ ansible_facts.distribution_major_version }}.yml"
+        - "{{ ansible_facts.os_family }}.yml"
+      skip: true
+  tags: always
+
+# we only override variables with our default if they have not been specified already.
+# by default the lookup functions finds all varnames containing the string, therefore
+# we add ^ and $ to denote start and end of string, so this returns only exact matches.
+- name: Set OS dependent variables, if not already defined by user # noqa var-naming
+  ansible.builtin.set_fact:
+    "{{ item.key }}": "{{ item.value }}"
+  when: not lookup('varnames', '^' + item.key + '$')
+  with_dict: "{{ os_vars }}"
+  tags: always
+
+#################################
+# Check Compatibility ###########
+#################################
+- name: Print the OS
+  ansible.builtin.debug:
+    var: ansible_facts.os_family
+
+- name: Only compatible OS versions
+  ansible.builtin.fail:
+    msg: "Only Ubuntu/Debian are supported"
+  when: ansible_facts.os_family not in ["Ubuntu", "Debian"]
+
+#################################
+# POSTGRES-01 ###################
+#################################
+- name: Check postgres service status
+  ansible.builtin.service:
+    name: "{{ postgres_daemon }}"
+    state: started
+
+#################################
+# POSTGRES-02 ###################
+#################################
+- name: Get postgres version
```

Comment:
Can we use the [postgresql_info_module](https://docs.ansible.com/ansible/latest/collections/community/postgresql/postgresql_info_module.html) here? According to the docs, getting the version is supported by the module.

Or don't you want to do this because then we'd have to connect to the postgres?

Reply:
@professormahi What do you think about this?